transformer_engine/pytorch/attention.py

```diff
@@ -2579,6 +2601,16 @@ class MultiheadAttention(torch.nn.Module):
           The device on which the parameters of the model will allocated. It is the user's
           responsibility to ensure all parameters are moved to the GPU before running the
           forward pass.
+    qkv_format: str, default = `sbhd`
+            dimension format for `query_layer`, `key_layer` and `value_layer`,
+            {`sbhd`, `bshd`, `thd`}. `s` stands for the sequence length, `b` batch size,
```

Comment:
So with this PR, our `TransformerLayer` will handle `sbhd`, `bshd` formats, and `MultiHeadAttention` and `DotProductAttention` will handle `sbhd`, `bshd`, `thd` formats?

Reply:
Ah and that's a gap right? I think we could provide `TransformerLayer` with `thd` as well. The `TransformerLayer` just passes that argument to the `MultiHeadAttention` layer anyways without doing anything with it itself.

Reply:
@cyanguwa should I add `thd` format to the docstring?

Reply:
I guess you shouldn't because TransformerLayer doesn't support `thd`. I think it's ok to keep `qkv_format = {sbhd, bshd}` for TransformerLayer and MultiHeadAttention, and `{sbhd, bshd, thd}` for DPA.

Reply:
sounds good!